### PR TITLE
feat/fix add query params to filter notes and search notes 

### PR DESCRIPTION
**Pull request by karthik01455**
fix: made one table component and one table header and reused it for 3 tables (action items , key decisions and agenda items)
feat: add query params in get API call for filters based on status, start date and end date.
feat: add query params in get API call for searching text in PO notes.
feat: add count of items in each table
fix: fetching made in tables instead of cards
fix: refactored add PO notes based on constraints added in backend for status.
fix: icon size, font family and layout
fix: rename folder structures to Camel case and filenames in Pascal case

 
- Filters and search based functionality are enabled from table so that user can search PO notes based on text and filter PO notes based on dates.
- One table is made and reused it to increase re usability.
- Fetching done in tables to make fetch and query params efficient.

image
note: rename folder structures to Camel case and filenames in Pascal case leads to creation of new files

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -23,7 +23,7 @@
       work correctly both with client-side routing and a non-root public URL.
       Learn how to configure a non-root public URL by running `npm run build`.
     -->
-  <title>React App</title>
+  <title>Agile Dashboard</title>
```
Comment (sugumar-mani) [inline public/index.html:26]: My Agile Dashboard

File: src/Components/Cards/theme.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+      main: '#08A0F7',
+      light: '#E6EEF2',
+      contrastText: '#FFFFFF',
+    },
+    secondary: {
+      main: '#707070',
+      light: '#E0E0E0',
+      contrastText: '#FFFFFF',
+    },
+
```
Comment (sugumar-mani) [inline src/Components/Cards/theme.js:15]: unused space

File: src/Components/Elements/NavBar.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
+import React, { useState } from 'react';
+import { Link, useLocation } from 'react-router-dom';
+import {
+  AppBar, Box, Toolbar, IconButton,
+  ThemeProvider, Typography, Menu,
+  Container, Avatar, Tooltip, MenuItem
+}
+  from '@mui/material';
```
Comment (sugumar-mani) [inline src/Components/Elements/NavBar.js:8]: seems like alignment issue

File: src/Components/PONotesComponents/AddPoNotes.js [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+  const [submit, setSubmit] = useState(false);
+  const [noteType, setNoteType] = useState('ACTION_ITEM');
+  const [statement, setStatement] = useState('');
+  const [timeline, setTimeline] = useState(getNextDate());
+  const handleNoteOpener = () => {
+    setAddNote(!addNote);
+  };
+
+  const getStatus = (type, status) => {
+    if (noteType === 'KEY_DECISION' && status !== "DRAFT") return "NONE"
```
Comment (sugumar-mani) [inline src/Components/PONotesComponents/AddPoNotes.js:36]: inconsistent use of single and double quotes

File: src/Components/PONotesComponents/PONotesTables/PONotesGridLayout.js [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+        <PONotesTable heading={agendaItems.heading}
+          definition={agendaItems.definition}
+          accessibilityInformation={agendaItems.accessibilityInformation}
+          query={query} />
+      </Grid>
+      <Grid item sx={1} />
+    </Grid>
+  );
+}
+
```
Comment (sugumar-mani) [inline src/Components/PONotesComponents/PONotesTables/PONotesGridLayout.js:83]: empty lines

File: src/Components/PONotesComponents/PONotesTables/PONotesTable.js [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+import { ThemeProvider } from '@mui/material/styles';
+import PropTypes from 'prop-types';
+import { useQuery } from 'react-query';
+import PONotsTableHeader from './PONotesTablesHeader/PONotesTableHeader';
+import PONotesTableTheme from '../../theme/GlobalTheme';
+import CardLayout from '../../cards/CardLayout';
+import { DOMAIN } from '../../../config';
+
+// query params for get Api call
+
```
Comment (sugumar-mani) [inline src/Components/PONotesComponents/PONotesTables/PONotesTable.js:21]: empty line

File: src/Components/PONotesComponents/PONotesTables/PONotesTable.js [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+            <TableBody >
+              <TableRow>
+                {/* Data from get Api call using query params is passed to cardlayout for displaying it in cards */}
+                <CardLayout chckBox type={map[heading]} data={data} /> </TableRow>
+            </TableBody>
+          </Table>
+        </TableContainer>
+      </ThemeProvider>
+    </Box>
+
```
Comment (sugumar-mani) [inline src/Components/PONotesComponents/PONotesTables/PONotesTable.js:92]: empty line. please check in all places

File: src/Components/Routes/Home.js [hunk 8]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import { Box } from '@mui/material';
+
+export default function HomeContainer() {
+  return (
+    <Box>
+      Welcome to Home page.
```
Comment (sugumar-mani) [inline src/Components/Routes/Home.js:7]: These static text will normally fed from internationalisation/localization in any client projects. Feel free to adopt internationalisation as you learn from your code academy in future.

File: src/Components/Theme/GlobalTheme.js [hunk 9]
```diff
@@ -0,0 +1,60 @@
+import { createTheme } from '@mui/material/styles';
+import '@fontsource/roboto/400.css';
+
+
```
Comment (sugumar-mani) [inline src/Components/Theme/GlobalTheme.js:4]: extra space

File: src/Components/UtilityFunctions/ErrorSnackbar.js [hunk 10]
```diff
@@ -0,0 +1,30 @@
+import { React, useState } from 'react';
+import Stack from '@mui/material/Stack';
+import Snackbar from '@mui/material/Snackbar';
+import MuiAlert from '@mui/material/Alert';
+import PropTypes from 'prop-types';
+
+export default function ErrorSnackbar({ message, setError }) {
+  const [open, setOpen] = useState(true);
+  const handleClose = (event, reason) => {
+    if (reason === 'clickaway') {
```
Comment (sugumar-mani) [inline src/Components/UtilityFunctions/ErrorSnackbar.js:10]: raw string can be declare in constants

**Review by sugumar-mani — APPROVED** [review]
Please feel free to go ahead if you are planning to fix the issues in future PRs.
Also remember we are going to have a single page app where the content scroll and not going to navigate among.

**Comment by karthik01455** [comment]
Incorporated all the changes. Raw strings will be declared as constants in all files and will be visible in future PRs.

**Pull request closed without merging**